xarray/core/dataset.py

```diff
@@ -10338,3 +10338,19 @@ def resample(
             restore_coord_dims=restore_coord_dims,
             **indexer_kwargs,
         )
+
+    def drop_attrs(self) -> Self:
+        """
+        Removes all attributes from the Dataset and its variables.
+        """
+        # Remove attributes from the dataset
+        self = self._replace(attrs={})
+
+        # Remove attributes from each variable in the dataset
+        for var in self.variables:
+            # variables don't have a `._replace` method, so we copy and then remove. If
+            # we added a `._replace` method, we could use that instead.
+            self[var] = self[var].copy()
```

Comment:
I think `_replace` already copies the variables.

Reply:
Hmmm, they're not copied deeply enough such that `self[var].attrs = {}` doesn't affect the original object — commenting out this line causes a failure at `assert_identical(ds, original)`...